### PR TITLE
Implement node API for handling IPC calls to codius API modules and other TLAs

### PR DESCRIPTION
**Pull request by tdfischer**

### CODE REVIEW

File: test/ipc.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+public:
+  void testSyncCall() {
+    codius_request_t* req;
+    codius_result_t* result;
+
+    pthread_t thread;
+    dup2 (test_fd[FD_SEND], 3);
+    pthread_create (&thread, NULL, IPCMessagingTest::listenThread, this);
+
+    req = codius_request_new ("test_api", "test_method");
```
Comment (wilsonianb) [inline test/ipc.cpp:37]: Should req be codius_request_free'd?

File: test/ipc.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    CPPUNIT_ASSERT_EQUAL (req->data, result->data);
+  }
+
+  void testReply() {
+    codius_request_t* req;
+    codius_request_t* sent_req;
+    codius_result_t* result;
+    codius_result_t* sent_result;
+
+    req = codius_request_new ("test_api", "test_method");
```
Comment (wilsonianb) [inline test/ipc.cpp:53]: Should req be codius_request_free'd?

Comment (tdfischer) [inline test/ipc.cpp:53]: Pushed a fix.

**Comment by wilsonianb** [comment]
LGTM :+1:

**Pull request merged**